src/silx/gui/plot/items/image.py

```diff
@@ -535,8 +535,10 @@ def setData(self, data, copy=True):
                           False to use internal representation (do not modify!)
         """
         data = numpy.array(data, copy=copy)
-        assert data.ndim == 3
-        assert data.shape[-1] in (3, 4)
+        if not data.ndim == 3:
+            raise ValueError(f"RGB(A) image expect to get 3D data. Gets {data.ndim}")
+        if not data.shape[-1] in (3, 4):
+            raise ValueError(f"RGB(A) image expect data[-1] to get 3 or 4 elements. Gets {data.shape[-1]}")
```

Comment:
Test simplifications:
```suggestion
        if data.ndim != 3:
            raise ValueError(f"RGB(A) image expect to get 3D data. Gets {data.ndim}")
        if data.shape[-1] not in (3, 4):
            raise ValueError(f"RGB(A) image expect data[-1] to get 3 or 4 elements. Gets {data.shape[-1]}")
```